conda/testing/helpers.py

```diff
@@ -220,16 +220,24 @@ def add_feature_records_legacy(index):
         index[rec] = rec
 
 
-def _export_subdir_data_to_repodata(subdir_data, index):
+def _export_subdir_data_to_repodata(subdir_data: SubdirData, index, subdir=None):
     """
     This function is only temporary and meant to patch wrong / undesirable
     testing behaviour. It should end up being replaced with the new class-based,
     backend-agnostic solver tests.
     """
     state = subdir_data._internal_state
+    subdir = subdir or subdir_data.channel.subdir
     packages = {}
     for pkg in index:
         data = pkg.dump()
+        if getattr(pkg, "noarch", None):
+            if subdir == "noarch":
+                data["subdir"] = "noarch"
+                data["platform"] = data["arch"] = None
+            else:
+                # skip noarch packages incorrectly labeled with a platform specific subdir
+                continue
```

Comment:
That's a little concerning, as it papers over the problem?

Reply:
I'll check which ones are problematic and then maybe we can talk about fixing that metadata in this PR or another.